### ISSUE TITLE
[FEATURE]: Allow glob matching in the `files` directive of the Kubernetes module

### ISSUE THREAD

**Issue by eysi09** [garden-io/garden#3465]
### Background / Motivation
Let's say I have the following project structure:

```
.
├── api
│   ├── garden.yml
│   ├── manifests
│   │   ├── Deployment.yaml
│   │   ├── Ingress.yaml
│   │   └── Service.yaml
│   └── src
└── project.garden.yml
```

If the API module has type `kubernetes` I want to be able to reference the manifests via glob matching as opposed to having to list all of them.

That is, this:

```yaml
kind: module
type: kubernetes
name: api
files: [./manifests/*]
```

as opposed to this:

```yaml
kind: module
type: kubernetes
name: api
files: [./manifests/Deployment.yaml, ./manifests/Ingress.yaml, ./manifests/Service.yaml]
```

### How important is this feature for you/your team?

:rose: It’s a nice to have, but nice things are nice :slightly_smiling_face:

**Comment by vvagaytsev** [comment]
It looks like this was fixed in #4516 (see https://github.com/garden-io/garden/pull/4516/commits/c9efb473d4f352ac13f12a7f8ada2a2b28445cb3).

We decided to revert #4516 in 0.13.8 as a hotfix for #4811. But there is #4814 to keep the safe parts of #4516 in 0.13.8.

**Comment by vvagaytsev** [comment]
Fixed in `0.13.7` via #4615 (see https://github.com/garden-io/garden/pull/4516/commits/c9efb473d4f352ac13f12a7f8ada2a2b28445cb3), retained in `0.13.8` via #4814 (see https://github.com/garden-io/garden/pull/4814/commits/05a43ce643d6d1cf99613504d044e5cd395d2dca).

**Comment by hnicke** [comment]
The [docs](https://docs.garden.io/kubernetes-plugins/action-types/kubernetes) should probably get updated.
They still say:
> Due to a current limitation you need to list all the manifests. There's an [open issue](https://github.com/garden-io/garden/issues/3465) for addressing this.

**Comment by vvagaytsev** [comment]
Thanks, @hnicke! 👍 I'll take a look.

**Comment by vvagaytsev** [comment]
@hnicke this will be fixed in #4879 and released in `0.13.10`.
